src/core/thread/tmf.cpp

```diff
@@ -164,11 +164,11 @@ bool Agent::HandleResource(const char *aUriPath, Message &aMessage, const Ip6::M
         Case(kUriAnycastLocate, AnycastLocator);
 #endif
 
-#if OPENTHREAD_FTD || OPENTHREAD_CONFIG_TMF_NETWORK_DIAG_MTD_ENABLE
-        Case(kUriDiagnosticGetRequest, NetworkDiagnostic::NetworkDiagnostic);
-        Case(kUriDiagnosticGetQuery, NetworkDiagnostic::NetworkDiagnostic);
-        Case(kUriDiagnosticGetAnswer, NetworkDiagnostic::NetworkDiagnostic);
-        Case(kUriDiagnosticReset, NetworkDiagnostic::NetworkDiagnostic);
+#if OPENTHREAD_CONFIG_TMF_NETDIAG_CLIENT_ENABLE
+        Case(kUriDiagnosticGetRequest, NetworkDiagnostic::Server);
```

Comment:
These are server-side handlers. Why are they guarded by `CLIENT_ENABLE`?

Reply:
Thanks. That's not correct. I missed this. Will fix in new push.